terraform/codebnb.tf

```diff
@@ -0,0 +1,338 @@
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.code_bnb.arn}"
+  principal     = "apigateway.amazonaws.com"
+
+  # More: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+  #source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.code_bnb.id}/*/${aws_api_gateway_method.add_candidate.http_method}${aws_api_gateway_resource.add_candidate.path}"
+  source_arn = "${aws_api_gateway_rest_api.code_bnb.execution_arn}/*/*"
+}
+
+resource "aws_lambda_permission" "apigw_lambda_add_candidate_permission" {
+  statement_id  = "AllowExecutionFromAPIGatewayAddCandidate"
+  action        = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.code_bnb.arn}"
+  principal     = "apigateway.amazonaws.com"
+
+  # More: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+  source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.code_bnb.id}/*/${aws_api_gateway_method.add_candidate.http_method}${aws_api_gateway_resource.add_candidate.path}"
+  #source_arn = "${aws_api_gateway_rest_api.code_bnb.execution_arn}/*/*"
+}
+
+resource "aws_lambda_permission" "apigw_lambda_post_assignment_permission" {
+  statement_id  = "AllowExecutionFromAPIGatewayPostAssignment"
+  action        = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.code_bnb.arn}"
+  principal     = "apigateway.amazonaws.com"
+
+  # More: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+  source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.code_bnb.id}/*/${aws_api_gateway_method.post_assignment.http_method}${aws_api_gateway_resource.assignment.path}"
+  #source_arn = "${aws_api_gateway_rest_api.code_bnb.execution_arn}/*/*"
+}
+
+resource "aws_lambda_permission" "apigw_lambda_get_assignment_permission" {
+  statement_id  = "AllowExecutionFromAPIGatewayGetAssignment"
+  action        = "lambda:InvokeFunction"
+  function_name = "${aws_lambda_function.code_bnb.arn}"
+  principal     = "apigateway.amazonaws.com"
+
+  # More: http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+  source_arn = "arn:aws:execute-api:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.code_bnb.id}/*/${aws_api_gateway_method.get_assignment.http_method}${aws_api_gateway_resource.assignment.path}"
+  #source_arn = "${aws_api_gateway_rest_api.code_bnb.execution_arn}/*/*"
+}
+
+# TODO: Figure out what demo is
```

Comment:
???  You never refer to the name of this particular resource, so you can call it whatever you want.  Or is your question: "Figure out why we're provisioning an API gateway account in the first place"?

Reply:
Both, I guess. (originally intended to question the name of the resource, but now yeah unclear what an API gateway account is to me). Leaving as a TODO, maybe someone will contribute a patch -- I know this applied successfully to my AWS account but not 100% confident that if I remove and re-install without the API gateway account that there won't be some pre-existing state as a result. If CodeBnb gets adoption we'll let others help with their experiences applying.